### PR TITLE
use shinymatrix and brew and update function df2graph

### PR DESCRIPTION
**Pull request by XDeng0921**

### CODE REVIEW

**Comment by nanxstats** [comment]
There seems to be some merge conflicts after the target branch is changed... could you discuss and resolve them? Overall this looks good to me.

File: modules/hgraph.R [hunk 1]
```diff
@@ -133,10 +133,10 @@ hGraph <- function(nHypotheses = 4, # Number of hypotheses
 
   indices <- 1:nHypotheses # make sure each ellipses's points are in its own "group" - different from the hypotheses groups used to color the ellipses (fill)
   ellipses <- rbind(
-    data.frame(x = rr$x, y = rr$y + h, fill = groupNames, fillalpha = fillalpha, group = indices),
-    data.frame(x = rr$x, y = rr$y - h, fill = groupNames, fillalpha = fillalpha, group = indices),
-    data.frame(x = rr$x + w, y = rr$y, fill = groupNames, fillalpha = fillalpha, group = indices),
-    data.frame(x = rr$x - w, y = rr$y, fill = groupNames, fillalpha = fillalpha, group = indices)
+    data.frame(x = as.numeric(rr$x), y = as.numeric(rr$y) + h, fill = groupNames, fillalpha = fillalpha, group = indices),
```
Comment (allenzhuaz) [inline modules/hgraph.R:136]: can be done out of `hGraph()`

File: templates/call-hgraph.R [hunk 2]
```diff
@@ -0,0 +1,27 @@
+h <- hGraph(
+  nHypotheses     = <%=nrow(DFAll)%>,
+  nodeTextTop     = c(<%=paste0(paste0('"', DFAll[,1], '"'), collapse = ", ")%>),
```
Comment (allenzhuaz) [inline templates/call-hgraph.R:3]: will be updated using `gMCPmini::hGraph()` arguments

File: server/main.R [hunk 3]
```diff
@@ -1,275 +1,223 @@
-# rhandsontable outputs --------------------------------------------------------
+observeEvent(input$btn_hotHypotheses_addrow, {
```
Comment (allenzhuaz) [inline server/main.R:1]: refactored the main file with two split files

**Pull request merged**